lib/widgets/inbox.dart

```diff
@@ -377,17 +380,44 @@ class _DmItem extends StatelessWidget {
   }
 }
 
-class _StreamHeaderItem extends _HeaderItem {
+class StreamHeaderItem extends _HeaderItem {
   final Subscription subscription;
 
-  const _StreamHeaderItem({
+  /// This context is used to ensure the [_StreamSection] that encloses
+  /// the current [StreamHeaderItem] is visible after being collpased
+  /// through this [StreamHeaderItem] being used as a sticky header.
+  ///
+  /// A non-null value should be passed if this [StreamHeaderItem]
+  /// is used as a sticky header.
+  final BuildContext? parentContext;
```

Comment:
This is a concern that's common to both kinds of sections and their headers, so let's put it on the base class `_HeaderItem`.

Let's also call it `sectionContext`, rather than "parent". The exact sequence of ancestors is rather an implementation detail that can change over time… and in fact I think already in this version the context this points to isn't actually the direct "parent" of this widget in any sense. Better to have the name reflect the thing we actually need to be true of this context, which is that it's the one for the whole section this header corresponds to.

Reply:
Done.